domain/application/state/application_endpoint.go

```diff
@@ -94,38 +133,33 @@ func (st *State) insertApplicationEndpoints(ctx context.Context, tx *sqlair.TX,
 		return errors.Errorf("checking charm relation: %w", err)
 	}
 
-	// Update default space.
-	if err := st.updateDefaultSpace(ctx, tx, params.appID, params.bindings); err != nil {
-		return errors.Errorf("updating default space: %w", err)
-	}
```

Comment:
I believe that it technically results in a better set of abstractions to keep raise `updateDefaultSpace` outside of this method, but as I progressed through the change, I came to understand more why it was originally here.

For anyone lurking, the reason is that the handling of endpoint bindings up until now (i.e. in Mongo, and the client) was very messy. An app's default space (property of an application) is bundled together with the actual binding themselves (entities in their own right).

We model this much better in DQLite, but since we can't break the API contract come messiness is necessary. I think it's best to handle this as explicitly as possible though, hence why I have raised this step up.